lls/main.c

```diff
@@ -600,7 +600,7 @@ lls_proc(void *arg)
 		/* Read in packets on front and back interfaces. */
 		int num_tx;
 
-		if (hw_filter_eth_available(front)) {
+		if (lls_conf->rx_method_front & RX_METHOD_NIC) {
```

Comment:
Shouldn't we move `hw_filter_eth_available()` from `include/gatekeeper_net.h` to `lib/net.c`? This also helps us to verify that we haven't forgotten to review any spot.

Reply:
The LLS block still uses `hw_eth_filter_available()` to check whether it should use `RX_METHOD_NIC`.

However, we could instead determine whether to use `RX_METHOD_NIC` based on the return value of `ethertype_filter_add()`, or pass the RX method as a `uint8_t *` to `ethertype_filter_add()` so that it can set it if the ethertype filter is installed.

And then `hw_eth_filter_available()` can be put into `lib/net.c` as you suggested.

Reply:
Let's keep `hw_filter_eth_available()` the way it is right now since we are not targeting that kind of filter in this pull request. I see that in the future we will likely integrate the filters mechanism, but let's leave it in the future.

Nevertheless, can't we move `hw_filter_ntuple_available()` from `include/gatekeeper_net.h` to `lib/net.c`?

Reply:
With a minor logging change to `ggu/main.c`, yes it can be moved to `lib/net.c` (in the next commit).